### PR TITLE
fix: fixed bug in valid_assay_link and finished documentation for renaming a Features instance

### PR DESCRIPTION
**Pull request by cvanderaa**

### CODE REVIEW

File: R/Features-class.R [hunk 1]
```diff
@@ -274,8 +277,11 @@ rowDataNames <- function(object) {
...
 
 
-#' @exportMethod names<-
-#' @rdname MultiAssayExperiment-methods
+##' @rdname Features-class
+##' 
+##' @param value A character() with new name(s) for the assay(s) in `object`  
+##' 
+##' @exportMethod names<-
 setReplaceMethod("names", c("Features", "ANY"),
```
Comment (lgatto) [inline R/Features-class.R]: Check of `ANY` shouldn't be rather `character` here.

File: R/Features-class.R [hunk 2]
```diff
@@ -274,8 +277,11 @@ rowDataNames <- function(object) {
...
                  function(x, value) {
                      key_vals <- cbind(names(x), value)
```
Comment (lgatto) [inline R/Features-class.R]: check here that `length(value)` equals `length(x)` and that `value` are unique.

File: R/Features-class.R (hunk 1 above)
Comment (cvanderaa) [reply R/Features-class.R]: Done ! I replaced `"ANY"` by `"character"`. Note that an error is thrown now even for numerics. For example:
```r
data(feat1)
names(feat1) <- 1
```
throws:
```
Error in `names<-`(`*tmp*`, value = 1) : 
  'value' must be a character vectorin names(x) <- value
```
But in my opinion this is a good behavior.

File: R/Features-class.R (hunk 2 above)
Comment (cvanderaa) [reply R/Features-class.R]: Actually, an error is already thrown at the `MultiAssayExperiment` level. I added some unit tests to make sure we notice any change in behavior from later versions.

File: R/Features-class.R (hunk 1 above)
Comment (lgatto) [reply R/Features-class.R]: Indeed.

**Comment by lgatto** [comment]
While testing locally:
```r
* checking examples ... ERROR
Running examples in ‘Features-Ex.R’ failed
The error most likely occurred in:

> ### Name: Features
> ### Title: Quantitative MS Features
> ### Aliases: Features Features-class class:Features addAssay
> ###   dims,Features-method show,Features-method
> ###   [,Features,ANY,ANY,ANY-method [,Features,character,ANY,ANY-method
> ###   rowDataNames selectRowData names<-,Features-method
> 
> ### ** Examples
> 
> ## ------------------------
> ## An empty Features object
> ## ------------------------
> 
> Features()
Error in listData[[1L]] : subscript out of bounds
Calls: Features -> MatchedAssayExperiment -> is
Execution halted
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’
 ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
  ── 2. Error: aggregateFeatures: empty and errors (@test_aggregateFeatures.R#4)  
  subscript out of bounds
  Backtrace:
   1. testthat::expect_identical(Features(), aggregateFeatures(Features()))
   4. Features::Features()
   5. MultiAssayExperiment::MatchedAssayExperiment(...)
   6. methods::is(listData[[1L]], "MultiAssayExperiment")
  
  ══ testthat results  ═══════════════════════════════════════════════════════════
  [ OK: 151 | SKIPPED: 1 | WARNINGS: 3 | FAILED: 2 ]
  1. Error: empty Features (@test_Features.R#4) 
  2. Error: aggregateFeatures: empty and errors (@test_aggregateFeatures.R#4) 
  
  Error: testthat unit tests failed
  Execution halted
* checking for unstated dependencies in vignettes ... OK
```
Could you check, please.

**Comment by cvanderaa** [comment]
I will checkt this, but I still have your reviewing to commit. So I wil create a follow up PR with the update + the eventual bug fixes.

**Pull request merged**